pootle/apps/pootle_project/models.py

```diff
@@ -339,13 +337,16 @@ def resources(self):
         '''
         cursor = connection.cursor()
         cursor.execute(sql_query, [resources_path, resources_path])
-
         results = cursor.fetchall()
 
         # Flatten tuple and sort in a list
-        resources = list(reduce(lambda x,y: x+y, results))
-        resources.sort(key=get_path_sortkey)
+        resources = set()
+        for result in results:
+            s = result[0]
```

Comment:
I like it a lot.

J. Leclanche

2015-02-03 23:23 GMT+01:00 Julen Ruiz Aizpuru <notifications@github.com>:

> In pootle/apps/pootle_project/models.py
> <https://github.com/translate/pootle/pull/3572#discussion_r24045790>:
>
> >          results = cursor.fetchall()
> >
> >          # Flatten tuple and sort in a list
> > -        resources = list(reduce(lambda x,y: x+y, results))
> > -        resources.sort(key=get_path_sortkey)
> > +        resources = set()
> > +        for result in results:
> > +            s = result[0]
>
> Check this out julen@32990d9
> <https://github.com/julen/pootle/commit/32990d94e52d1101b4c6b656740a971ec9f4ffc0>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/translate/pootle/pull/3572/files#r24045790>.
>